transformer_engine/jax/cpp_extensions.py

```diff
@@ -4356,13 +4181,14 @@ def abstract(dz_aval, amax_aval, scale_aval, scale_inv_aval, *, out_dtype,
         assert amax_aval.dtype == jnp.float32
         assert scale_aval.dtype == jnp.float32
         assert scale_inv_aval.dtype == jnp.float32
-        gi_hidden_size = dz_aval.shape[-1]
+        if dz_aval.shape[-2] == 1 or dz_aval.shape[-2] == 2:
+            gi_hidden_size = reduce(operator.mul, dz_aval.shape[-2:])
+        else:
+            gi_hidden_size = dz_aval.shape[-1]
```

Comment:
When will dz_aval.shape[-2] != 1 or 2?

Reply:
Hi, the first `dbias_cast_transpose` in the `_fused_layernorm_fp8_mlp_bwd` is with `grad` input which has the size of `[batch_size, out]`. So its `shape[-2]` is neither 1 nor 2.

Reply:
For this cases, the code here is not safe. It is possible that the batch_size = 2, then the gi_hidden_size will be computed incorrectly. Can you try to use other metrics to determine if the input is gated? (I'm not sure but probably the transpose_axis_boundary)

Reply:
I pushed the change in which transpose_axis_boundary is used for conditional check. 
I left a note and added an error raise there, in case if it causes any issue in the future.